gctscript/modules/gct/exchange.go

```diff
@@ -0,0 +1,495 @@
+			funds.Value = append(funds.Value, &objects.Map{Value: temp})
+		}
+	}
+
+	data := make(map[string]objects.Object, 2)
+	data["exchange"] = &objects.String{Value: rtnValue.Exchange}
+	data["currencies"] = &funds
+
+	return &objects.Map{
+		Value: data,
+	}, nil
+}
+
+// ExchangeOrderQuery query order on exchange
+func ExchangeOrderQuery(args ...objects.Object) (objects.Object, error) {
+	if len(args) != 2 {
+		return nil, objects.ErrWrongNumArguments
+	}
+
+	exchangeName, ok := objects.ToString(args[0])
+	if !ok {
+		return nil, fmt.Errorf(ErrParameterConvertFailed, exchangeName)
+	}
+	orderID, ok := objects.ToString(args[1])
+	if !ok {
+		return nil, fmt.Errorf(ErrParameterConvertFailed, orderID)
+	}
+	orderDetails, err := wrappers.GetWrapper().QueryOrder(exchangeName, orderID)
+	if err != nil {
+		return nil, err
+	}
+
+	var tradeHistory objects.Array
+	for x := range orderDetails.Trades {
+		temp := make(map[string]objects.Object, 7)
+		temp["timestamp"] = &objects.Time{Value: orderDetails.Trades[x].Timestamp}
+		temp["price"] = &objects.Float{Value: orderDetails.Trades[x].Price}
+		temp["fee"] = &objects.Float{Value: orderDetails.Trades[x].Fee}
+		temp["amount"] = &objects.Float{Value: orderDetails.Trades[x].Amount}
+		temp["type"] = &objects.String{Value: orderDetails.Trades[x].Type.String()}
+		temp["side"] = &objects.String{Value: orderDetails.Trades[x].Side.String()}
+		temp["description"] = &objects.String{Value: orderDetails.Trades[x].Description}
+		tradeHistory.Value = append(tradeHistory.Value, &objects.Map{Value: temp})
+	}
+
+	data := make(map[string]objects.Object, 14)
+	data["exchange"] = &objects.String{Value: orderDetails.Exchange}
+	data["id"] = &objects.String{Value: orderDetails.ID}
+	data["accountid"] = &objects.String{Value: orderDetails.AccountID}
+	data["currencypair"] = &objects.String{Value: orderDetails.CurrencyPair.String()}
+	data["price"] = &objects.Float{Value: orderDetails.Price}
+	data["amount"] = &objects.Float{Value: orderDetails.Amount}
+	data["amountexecuted"] = &objects.Float{Value: orderDetails.ExecutedAmount}
+	data["amountremaining"] = &objects.Float{Value: orderDetails.RemainingAmount}
+	data["fee"] = &objects.Float{Value: orderDetails.Fee}
+	data["side"] = &objects.String{Value: orderDetails.OrderSide.String()}
+	data["type"] = &objects.String{Value: orderDetails.OrderType.String()}
+	data["date"] = &objects.String{Value: orderDetails.OrderDate.String()}
+	data["status"] = &objects.String{Value: orderDetails.Status.String()}
+	data["trades"] = &tradeHistory
+
+	return &objects.Map{
+		Value: data,
+	}, nil
+}
+
+// ExchangeOrderCancel cancels order on requested exchange
+func ExchangeOrderCancel(args ...objects.Object) (objects.Object, error) {
+	if len(args) != 2 {
+		return nil, objects.ErrWrongNumArguments
+	}
+
+	exchangeName, ok := objects.ToString(args[0])
+	if !ok {
+		return nil, fmt.Errorf(ErrParameterConvertFailed, exchangeName)
+	}
+	orderID, _ := objects.ToString(args[1])
```

Comment:
Please check the conversion here and thanks for adding these to all of the other ones